### PR TITLE
Change methods to return void when return value is not used

### PR DESCRIPTION
**Pull request by jwhui**
This PR contains a number of small commits that should make it easier to review.

More will come in subsequent PRs.

### CODE REVIEW

File: src/core/coap/coap.cpp [hunk 1]
```diff
@@ -478,12 +478,15 @@ otError CoapBase::SendCopy(const Message &aMessage, const Ip6::MessageInfo &aMes
 
 exit:
 
-    if (error != OT_ERROR_NONE && messageCopy != NULL)
+    if (error != OT_ERROR_NONE)
     {
-        messageCopy->Free();
-    }
+        otLogNoteCoap("Failed to send copy: %s", otThreadErrorToString(error));
```
Comment (simonlingoogle) [inline src/core/coap/coap.cpp:483]: otLogWarnCoap ?

File: src/core/meshcop/commissioner.cpp [hunk 2]
```diff
@@ -312,7 +316,7 @@ otError Commissioner::AddJoiner(const Mac::ExtAddress *aEui64, const char *aPskd
 
         UpdateJoinerExpirationTimer();
 
-        IgnoreError(SendCommissionerSet());
+        SendCommissionerSet();
```
Comment (simonlingoogle) [inline src/core/meshcop/commissioner.cpp:319]: If `SendCommissionerSet()` failed here, do we have any mechanism to assure that the Commissioner information is eventually synchronized to Leader?

File: src/core/thread/indirect_sender.cpp [hunk 3]
```diff
@@ -84,23 +84,22 @@ void IndirectSender::Stop(void)
...
 }
 
-otError IndirectSender::AddMessageForSleepyChild(Message &aMessage, Child &aChild)
+void IndirectSender::AddMessageForSleepyChild(Message &aMessage, Child &aChild)
 {
-    otError  error = OT_ERROR_NONE;
     uint16_t childIndex;
 
-    VerifyOrExit(!aChild.IsRxOnWhenIdle(), error = OT_ERROR_INVALID_STATE);
+    VerifyOrExit(!aChild.IsRxOnWhenIdle(), OT_NOOP);
```
Comment (simonlingoogle) [inline src/core/thread/indirect_sender.cpp:91]:
```suggestion
    OT_ASSERT(!aChild.IsRxOnWhenIdle());
```

File: src/core/thread/mle_router.cpp [hunk 4]
```diff
@@ -3264,8 +3252,6 @@ otError MleRouter::SendDataResponse(const Ip6::Address &aDestination,
     {
         message->Free();
```
Comment (simonlingoogle) [inline src/core/thread/mle_router.cpp:3253]: log fail here?

File: src/core/thread/mle_router.cpp [hunk 5]
```diff
@@ -467,8 +469,6 @@ otError MleRouter::SendAdvertisement(void)
...
```
Comment (simonlingoogle) [inline src/core/thread/mle_router.cpp:470]: log error?

File: src/core/meshcop/commissioner.cpp (hunk 2 above)
Comment (jwhui) [reply src/core/meshcop/commissioner.cpp]: Yes, that would require more changes and beyond the scope of this PR.

File: src/core/thread/mle_router.cpp [hunk 6]
```diff
@@ -374,6 +374,12 @@ void MleRouter::SetStateLeader(uint16_t aRloc16)
     otLogNoteMle("Leader partition id 0x%x", mLeaderData.GetPartitionId());
 }
 
+void MleRouter::AddLeaderAloc(void)
+{
+    IgnoreError(GetLeaderAloc(mLeaderAloc.GetAddress()));
+    IgnoreError(Get<ThreadNetif>().AddUnicastAddress(mLeaderAloc));
+}
```
Comment (abtink) [inline src/core/thread/mle_router.cpp:381]: I verified that we don't need `VerifyOrExit(IsLeader())`
Noticed it is being used only from `SetStateLeader()`. Can we consider removing this method (inlining it)?

File: src/core/thread/mle_router.cpp [hunk 7]
```diff
@@ -3260,12 +3253,15 @@ otError MleRouter::SendDataResponse(const Ip6::Address &aDestination,
 
 exit:
 
-    if (error != OT_ERROR_NONE && message != NULL)
+    if (error != OT_ERROR_NONE)
     {
-        message->Free();
-    }
+        otLogWarnMle("Failed to send Data Response: %s", otThreadErrorToString(error));
```
Comment (simonlingoogle) [inline src/core/thread/mle_router.cpp:3258]: Can we further reduce the code size by using same formatting string for all send failures?
```suggestion
        otLogWarnMle("Failed to send %s: %s", "Data Response", otThreadErrorToString(error));
```

File: src/core/thread/mle.cpp [hunk 8]
```diff
@@ -2508,8 +2494,6 @@ otError Mle::SendAnnounce(uint8_t aChannel, bool aOrphanAnnounce, const Ip6::Add
     {
         message->Free();
```
Comment (simonlingoogle) [inline src/core/thread/mle.cpp:2495]: log failure? 

Would it be better if we use the same statement for both successful log and failure log, like:
```suggestion

if (error == OT_ERROR_NONE) {
  otLogInfoMle("Send Announce on channel %d: %s", aChannel, otThreadErrorToString(error));
} else {
  otLogWarnMle("Send Announce on channel %d: %s", aChannel, otThreadErrorToString(error));
}

        message->Free();
```
We can define a new MACRO for this purpose.

Comment (jwhui) [reply src/core/thread/mle.cpp:2495]: We can consider this, but probably better for a separate PR that covers all instances that are not within scope of this PR.

Comment (simonlingoogle) [reply src/core/thread/mle.cpp:2495]: Agree

File: src/core/thread/mle_router.cpp (hunk 7 above)
Comment (jwhui) [reply src/core/thread/mle_router.cpp:3253]: We can consider this, but probably better for a separate PR that covers all instances that are not within scope of this PR.

**Pull request merged**